arangod/Aql/QueryRegistry.cpp

```diff
@@ -267,7 +308,10 @@ void QueryRegistry::closeEngine(EngineId engineId) {
       if (canDestroyQuery) {
         auto queryMapIt = _queries.find(ei._queryInfo->_query->id());
         TRI_ASSERT(queryMapIt != _queries.end());
-        deleteQuery(queryMapIt);
+        queryInfoLifetimeExtension = deleteQuery(queryMapIt);
+
+        postQueryDestructionTrackingTask(ei._queryInfo->_query->id(),
+                                         queryInfoLifetimeExtension);
```

Comment:
Should we maybe move this call outside of the lock, and similarly in the other call sites of it?